runtime/obj.c

```diff
@@ -80,20 +81,39 @@ CAMLprim value caml_obj_make_forward (value blk, value fwd)
   return Val_unit;
 }
 
+static struct custom_operations default_custom_ops = {
+  "_default_custom_ops",
+  custom_finalize_default,
+  custom_compare_default,
+  custom_hash_default,
+  custom_serialize_default,
+  custom_deserialize_default,
+  custom_compare_ext_default,
+  custom_fixed_length_default
+};
+
 /* [size] is a value encoding a number of blocks */
 CAMLprim value caml_obj_block(value tag, value size)
 {
   value res;
-  mlsize_t sz, i;
+  mlsize_t sz;
   tag_t tg;
 
   sz = Long_val(size);
   tg = Long_val(tag);
-  if (sz == 0) return Atom(tg);
-  res = caml_alloc(sz, tg);
-  for (i = 0; i < sz; i++)
-    Field(res, i) = Val_long(0);
+  if (sz == 0) {
+    res = Atom(tg);
+  } else {
+    res = caml_alloc(sz, tg);
+  }
```

Comment:
Is there any value in hoisting the `sz == 0` case?
```suggestion
  caml_alloc(sz, tg)
```

Reply:
(if not, there should really be a `CAMLassert (tag < 256)` somewhere)

Reply:
Fixed.